From: Matthias Clasen <mclasen@redhat.com>
Date: Wed, 7 Apr 2021 14:28:39 -0400
Subject: wayland: Improve font setting fallback more

We may get a response from the portal that contains
no useful settings at all. In that case, we should
fallback as well.

Bug: #3838
Origin: upstream, 3.24.29, commit:dfea8c31d97379ff62635474ffde0d388b4b85ac
---
 gdk/wayland/gdkscreen-wayland.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/gdk/wayland/gdkscreen-wayland.c b/gdk/wayland/gdkscreen-wayland.c
index e9a4f0e..6f8b5e8 100644
--- a/gdk/wayland/gdkscreen-wayland.c
+++ b/gdk/wayland/gdkscreen-wayland.c
@@ -745,6 +745,14 @@ init_settings (GdkScreen *screen)
           goto fallback;
         }
 
+      if (g_variant_n_children (ret) == 0)
+        {
+          g_debug ("Received no portal settings");
+          g_clear_pointer (&ret, g_variant_unref);
+
+          goto fallback;
+        }
+
       g_variant_get (ret, "(a{sa{sv}})", &iter);
 
       while (g_variant_iter_loop (iter, "{s@a{sv}}", &schema, &val))
